azure-devops/azext_devops/dev/common/telemetry.py

```diff
@@ -67,7 +67,9 @@ def _is_telemetry_enabled():
     az_config = az_cli_ctx.config
     if az_config.has_option('core', 'collect_telemetry'):
         collect_telemetry = az_config.get('core', 'collect_telemetry')
-    return bool(collect_telemetry is None or collect_telemetry != 'no')
+    elif az_config.has_option('defaults', 'collect_telemetry'):
+        collect_telemetry = az_config.get('defaults', 'collect_telemetry')
```

Comment:
I think the [documentation](https://docs.microsoft.com/en-us/cli/azure/azure-cli-configuration?view=azure-cli-latest#cli-configuration-values-and-environment-variables) suggests default should overwrite core. 
This code would do the reverse. 

This change is still prone to mismatch later if Azure CLI changes their logic or priority. The telemetry check should be given by Azure CLI as a public function to be consumed by the extensions so that the logic remains in one place. 

Lets follow up on this with Azure CLI before pushing this fix.